.NET/Microsoft.Recognizers.Text.DateTime/Parsers/CJK/BaseCJKDatePeriodParser.cs

```diff
@@ -291,6 +296,63 @@ public int GetYearFromText(Match match)
             return year;
         }
 
+        private static DateObject GetFirstThursday(int year, int month = Constants.InvalidMonth)
```

Comment:
Shoudn't these two methods be in a common palce that can be called everywhere? The calculations are not CJK specific.
Maybe this should be in some utility in the timexlib?

Reply:
moved to DateObjectExtension